hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieLogBlock.java

```diff
@@ -16,13 +16,17 @@
 
 package com.uber.hoodie.common.table.log.block;
 
+import com.uber.hoodie.common.table.log.block.metadata.HoodieLogBlockMetadata;
+
 import java.io.IOException;
+import java.util.List;
 
 /**
  * Abstract interface defining a block in HoodieLogFile
  */
 public interface HoodieLogBlock {
   byte[] getBytes() throws IOException;
+  List<HoodieLogBlockMetadata> getMetadata();
```

Comment:
Added this as part of the LogBlock, debated between returning byte [] or the constructed list of metadata, the latter felt more optimized since don't have to convert again and again.

Reply:
Thinking of changing List<> to Array [], thoughts ?

Reply:
yeah having a List or Array is fine.. Not following specifics of list vs array

Reply:
In general, returning primitive type (like an array) consuming only the space it needs that's all..